src/Workspaces/Core/Portable/Workspace/Solution/TextDocumentState.cs

```diff
@@ -107,6 +114,12 @@ protected static ValueSource<TextAndVersion> CreateRecoverableText(TextLoader lo
             }
             catch (IOException e)
             {
+                if (++retries <= MaxRetries)
```

Comment:
What about UnauthorizedAccessException? Or are we already wrapping all non-IOExceptions with IOException deep down in the text loader?